File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climb(self,idx,n,dp): 
        if(idx>n):
            return 0
        if(idx==n): 
            return 1
        if(dp[idx]!=-1): 
            return dp[idx] 
        dp[idx]=self.climb(idx+1,n,dp)+self.climb(idx+2,n,dp) 
        return dp[idx]
    def climbStairs(self, n: int) -> int:
        # each idx you can +1 or +2
        dp = [-1]*n 
        return self.climb(0,n,dp)

File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public: 
    int climbStairs(int n) {
        if(n<2) return 1;
        int second_prev=-1,prev=1;
        int curr=0;
        for(int i=1;i<=n;i++){
            curr=prev;
            if(i>1) curr+=second_prev;
            second_prev=prev;
            prev=curr;
        }
        return curr;
    }
};